### PR TITLE
Add instructions for deployment

### PR DESCRIPTION
**Pull request by manics**
Moves most of the README into setup.md. Adds deployment.md.

### CODE REVIEW

**Comment by jburel** [comment]
@manics thanks

File: docs/setup.md [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+
+## Install GitLab runner
+
+https://docs.gitlab.com/ee/install/kubernetes/gitlab_runner_chart.html#installing-gitlab-runner-using-the-helm-chart
+
+Go to the `Runners settings` tab under https://gitlab.com/USERNAME/REPOSITORY/settings/ci_cd.
+Get the Runner registration token from and set `runnerRegistrationToken` in gitlab-runner/helm-config.yaml to this value.
+The gitlab runner requires additional configuration that is not supported by the chart so at present you must [checkout a fork/PR](https://gitlab.com/charts/charts.gitlab.io/merge_requests/120):
+
+    git clone --branch runner-environment https://gitlab.com/manics/charts.gitlab.io.git
```
Comment (joshmoore) [inline docs/setup.md]: This will need migrating.

File: docs/deployment.md [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+  - [`charts/jupyterhub-shared-scratch`](../charts/jupyterhub-shared-scratch) creates the shared scratch space for the VAE (user scratch spaces are automatically created by JupyterHub)
+
+
+## Configuration
+
+There are four JupyterHub deployments.
+All use the common configuration in [`jupyterhub-config.yaml`](../jupyterhub-config.yaml) with additional configuration or overrides in separate files.
+
+### 1. JupyterHub `staging`
+This is always deployed as soon as the `master` branch is updated, and is fully configured by [`jupyterhub-config.yaml`](../jupyterhub-config.yaml).
```
Comment (joshmoore) [inline docs/deployment.md:27]: Could see this being renamed to `-staging.yaml`

**Review by joshmoore — DISMISSED** [review]
One minor change request. Other than that, looks good and is understandable. Optimal would be using this to create a `k8s-test-deploy` (incl. gitlab repo, etc) to be sure, but I haven't done that yet.

**Comment by jburel** [comment]
The runners used were no longer available. So it was necessary to create a runner
It was also necessary to use a newer version of gitlab-runner (0.9.1)
* to add repo: ```helm repo add gitlab https://charts.gitlab.io```
* to update: ```helm repo update```
* update gitlab runner config file. PR to be opened
* To add new runner go to Settings/runners on gitlab, copy the token and paste in https://github.com/IDR/k8s-analysis-deploy/blob/98073e673eebc6e15a422ecaecd6778ab073353a/gitlab-runner/helm-config.yaml#L5
* Run 
```
helm upgrade --install --namespace gitlab gitlab-runner gitlab/gitlab-runner --version 0.9.1 -f /Users/jmarie/Documents/git-repo/k8s-analysis-deploy/gitlab-runner/helm-config.yaml 
```

File: docs/deployment.md (hunk 2 above)
Comment (manics) [reply docs/deployment.md:27]: This is part of the standard config used by all jupyterhub deployments, the rest add to it.

**Comment by manics** [comment]
Updated

**Comment by jburel** [comment]
gitlab-runner has been upgraded, the requested change by @joshmoore no longer applies (marking it as dismiss)
Thanks @manics for updating the instruction
Merging

**Pull request merged**